### ISSUE TITLE
Support  torch QAT model export? 

### ISSUE THREAD

**Issue by WangFengtu1996** [google-ai-edge/ai-edge-torch#178]
### Description of the Feature:

torch QAT supports three mode.
* Eager Mode Quantization
* FX Graph Mode Quantization 
* PyTorch 2 Export Quantization
detail in https://pytorch.org/docs/stable/quantization.html#module-torch.ao.quantization.backend_config.executorch

which mode  is supprted by `ai-edge-torch` ? 
thkx

**Comment by pkgoogle** [comment]
Hi @WangFengtu1996, we actively working on supporting more modes/recipes .. currently for PT Converter we support P2TE and original TFLite quantization: https://github.com/google-ai-edge/ai-edge-torch/blob/main/docs/pytorch_converter/README.md#quantization

For the generative API we support this: https://github.com/google-ai-edge/ai-edge-torch/tree/main/ai_edge_torch/generative/quantize

Does that answer your question?

**Comment by pkgoogle** [comment]
I am unsure if that is fully supported... I have found some resources which I think you can combine w/ the original TFLite Quantization above:

https://www.tensorflow.org/model_optimization/guide/quantization/training

You may need to change some steps to adjust for this workflow (which didn't exist at the time of the above documentation writing). Like you may want to switch to keras3 with PyTorch backend in the above steps and then attempt to use original TFLite Quantization.

**Comment by killamocingbird** [comment]
+1 on this question, if I do QAT can I then export and expect it to work? When I export after QAT I see a huge accuracy drop
```python
    model_for_export = get_model(
        model_name=model_name,
        n_classes=n_classes,
        input_size=input_size,
        **model_kwargs,
    )
    model_for_export.eval()
    model_for_export = ai_edge_torch.to_channel_last_io(model_for_export, args=[0])
    sample_inputs = (torch.randn(1, input_size, input_size, 3),)
    pt2e_export = export_for_training(model_for_export, sample_inputs).module()
    pt2e_export = quantize_pt2e.prepare_qat_pt2e(pt2e_export, quantizer)

    if qat_best_path.exists():
        logger.info("Loading best QAT weights before conversion...")
        pt2e_export.load_state_dict(torch.load(qat_best_path, weights_only=True))
        # Clean up temp file
        qat_best_path.unlink()
    else:
        raise FileNotFoundError(f"QAT best weights not found: {qat_best_path}")

    logger.info("Moving model to CPU for conversion...")
    pt2e_export = pt2e_export.to("cpu")
    pt2e_export = move_exported_model_to_eval(pt2e_export)

    logger.info("Converting to quantized PyTorch model...")
    pt2e_export = quantize_pt2e.convert_pt2e(pt2e_export, fold_quantize=False)
    pt2e_export = move_exported_model_to_eval(pt2e_export)

    # Export to TFLite with quantization config
    logger.info("Exporting to TFLite format with int8 quantization...")
    sample_inputs = (torch.randn(1, input_size, input_size, 3),)
    tflite_model = ai_edge_torch.convert(
        pt2e_export,
        sample_inputs,
        quant_config=quant_config.QuantConfig(pt2e_quantizer=quantizer),
    )
```
